hack/save-logs-files.sh

```diff
@@ -0,0 +1,237 @@
+            for ns in ${namespaces_list[@]}; do
+                if [[ ! " ${existant_ns[@]} " =~ " ${ns} " ]]; then
+                    echo "error: given namespace '$ns' doesn't exist"
+                    print_usage
+                    exit 1
+                fi
+            done
+            ;;
+        $FLAG_PATH)
+            output_path=$(readlink -f "$OPTARG")/
+            if [[ ! -d $output_path ]]; then
+                echo "error: given path '$output_path' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            ;;
+        $FLAG_APP)
+            application_args_list=()
+            readarray -d $ARGS_SEPARATOR -t application_args_list < <(printf '%s' "$OPTARG")
+            if [ ${#application_args_list[@]} -ne 2 ] || [ -z ${application_args_list[0]} ] || [ -z ${application_args_list[1]} ]; then
+                echo "error: given application namespace and name should be separated by a single comma"
+                print_usage
+                exit 1
+            fi
+            existant_ns=$(kubectl get ns -o=jsonpath='{.items[*].metadata.name}')
+            fybrik_application_ns=${application_args_list[0]}
+            fybrik_application_name=${application_args_list[1]}
+            if [[ ! " ${existant_ns[@]} " =~ " ${fybrik_application_ns} " ]]; then
+                echo "error: given namespace '$fybrik_application_ns' in flag '$FLAG_APP' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            fybrik_applications_in_ns=$(kubectl get fybrikapplications -n $fybrik_application_ns -o=jsonpath='{.items[*].metadata.name}')
+            if [[ ! " ${fybrik_applications_in_ns[@]} " =~ " ${fybrik_application_name} " ]]; then
+                echo -e "error: given fybrikapplication '$fybrik_application_ns:$fybrik_application_name' doesn't exist"
+                print_usage
+                exit 1
+            fi
+            ;;
+        $FLAG_UUID)
+            fybrik_application_uuid="$OPTARG"
+            ;;
+        $FLAG_HELP)
+            print_description
+            print_usage
+            print_flags
+            exit 0
+            ;;
+        : | ?)
+            print_usage
+            exit 1
+            ;;
+    esac
+done
+shift "$(($OPTIND -1))"
+
+# if no namespaces were provided, set value to default namespaces
+if [[ ${#namespaces_list[@]} -eq 0 ]]; then
+    for ns in $(kubectl get ns -o=jsonpath='{.items[*].metadata.name}'); do
+        if [[ $ns == *"fybrik"* ]]; then
+            namespaces_list+=("$ns")
+        fi
+    done
+fi
+
+# create a directory for saving output
+dirname_logs=$output_path"fybrik_logs_$(date +%Y-%m-%d_%H-%M)_last_$duration"
+echo "Output will be saved in $dirname_logs"
+mkdir $dirname_logs
+dirname_logs_by_container=$dirname_logs/logs_by_container
+mkdir $dirname_logs_by_container
+
+relevant_configmaps=("cluster-metadata" "fybrik-config")
```

Comment:
Is it possible to print config-maps of the deployed modules in fybrik-blueprints namespace? and also the logs of the deployed modules in that namespace.

Reply:
Regarding adding configmaps , I will add to the script all the non-certificate configmaps in the explored namespaces (including `fybrik-blueprints` namespace as it is explored by default).
The logs of the pods in the `fybrik-blueprints` namespace are already extracted, including from the pods that are connected to the modules.